pkg/machinery/config/encoder/documentation.go

```diff
@@ -239,7 +264,7 @@ func readExample(v reflect.Value, doc *Doc, index int) {
 		index = numExamples - 1
 	}
 
-	defaultValue := reflect.ValueOf(doc.Examples[index].Value)
+	defaultValue := reflect.ValueOf(doc.Examples[index].value)
```

Comment:
```suggestion
	defaultValue := reflect.ValueOf(doc.Examples[index].GetValue())
```

Reply:
nope, that's causing a deadlock. As we surround `populateExamples` with `Lock` `Unlock` already.
Not particularly like this solution though.

Reply:
Uh!

Reply:
yeah, I feel I need more thorough refactoring here. Doing that now...

Reply:
now it should become slightly better.